pyOCD/tools/pyocd.py

```diff
@@ -69,207 +190,514 @@ def dumpHexData(data, startAddress=0, width=8):
+        else:
+            data = [self.convert_value(d) for d in args[1:]]
+
+        if self.args.width == 8:
+            pass
+        elif self.args.width == 16:
+            data = pyOCD.utility.conversion.half2byte(data)
+        elif self.args.width == 32:
+            data = pyOCD.utility.conversion.word2byte(data)
+
+        self.target.writeBlockMemoryUnaligned8(addr, data)
+
+    def handle_erase(self, args):
+        self.flash.init()
+        self.flash.eraseAll()
+
+    def handle_unlock(self, args):
+        # Currently the same as erase.
+        if not self.didErase:
+            self.target.massErase()
+
+    def handle_go(self, args):
+        self.target.resume()
+        status = self.target.getState()
+        if status == pyOCD.target.cortex_m.TARGET_RUNNING:
+            print "Successfully resumed device"
+        else:
+            print "Failed to resume device"
+
+    def handle_step(self, args):
+        self.target.step()
+        print "Successfully stepped device"
+
+    def handle_halt(self, args):
+        self.target.halt()
+
+        status = self.target.getState()
+        if status != pyOCD.target.cortex_m.TARGET_HALTED:
+            print "Failed to halt device"
+            return 1
+        else:
+            print "Successfully halted device"
+
+    def handle_log(self, args):
+        if len(args) < 1:
+            print "Error: no log level provided"
+            return 1
+        if args[0].lower() not in LEVELS:
+            print "Error: log level must be one of {%s}" % ','.join(LEVELS.keys())
+            return 1
+        logging.getLogger().setLevel(LEVELS[args[0].lower()])
+
+    def handle_clock(self, args):
+        if len(args) < 1:
+            print "Error: no clock frequency provided"
+            return 1
+        try:
+            freq_Hz = int(args[0]) * 1000
+        except:
+            print "Error: invalid frequency"
+        self.transport.setClock(freq_Hz)
+
+        if self.transport.mode == pyOCD.transport.cmsis_dap.DAP_MODE_SWD:
+            swd_jtag = 'SWD'
+        else:
+            swd_jtag = 'JTAG'
+
+        if freq_Hz >= 1000000:
+            nice_freq = "%.2f MHz" % (freq_Hz / 1000000)
+        elif freq_Hz > 1000:
+            nice_freq = "%.2f kHz" % (freq_Hz / 1000)
+        else:
+            nice_freq = "%d Hz" % freq_Hz
+
+        print "Changed %s frequency to %s" % (swd_jtag, nice_freq)
+
+    def handle_exit(self, args):
+        raise ToolExitException()
+
+    ## @brief Convert an argument to a 32-bit integer.
+    #
+    # Handles the usual decimal, binary, and hex numbers with the appropriate prefix.
+    # Also recognizes register names and address dereferencing. Dereferencing using the
+    # ARM assembler syntax. To dereference, put the value in brackets, i.e. '[r0]' or
+    # '[0x1040]'. You can also use put an offset in the brackets after a comma, such as
+    # '[r3,8]'. The offset can be positive or negative, and any supported base.
```

Comment:
Are you using doxygen syntax here?  Python has built in support for documentation strings that might better to use instead.  More info on this can be found in pep-0257.  Probably not worth changing now, but I figured I would mention it for future reference.